iteratee/src/main/scala/io/finch/iteratee/package.scala

```diff
@@ -15,82 +14,38 @@ import shapeless.Witness
 package object iteratee extends IterateeInstances {
 
 
-  private[finch] def enumeratorFromReader[F[_] : Effect](reader: Reader[Buf]): Enumerator[F, Buf] = {
-    def rec(reader: Reader[Buf]): Enumerator[F, Buf] = {
-      Enumerator.liftM[F, Option[Buf]] {
-        futureToEffect(reader.read())
-      }.flatMap {
-        case None => Enumerator.empty[F, Buf]
-        case Some(buf) => Enumerator.enumOne[F, Buf](buf).append(rec(reader))
-      }
-    }
-    rec(reader).ensure(Effect[F].delay(reader.discard()))
-  }
-
-  /**
-    * An evaluating [[Endpoint]] that reads a required chunked streaming binary body, interpreted as
-    * an `Enumerator[Future, A]`. The returned [[Endpoint]] only matches chunked (streamed) requests.
-    */
-  def enumeratorBody[F[_] : Effect, A, CT <: String](implicit
-    decode: Enumerate.Aux[F, A, CT]
-  ): Endpoint[F, Enumerator[F, A]] = new Endpoint[F, Enumerator[F, A]] {
-      final def apply(input: Input): Endpoint.Result[F, Enumerator[F, A]] = {
-        if (!input.request.isChunked) EndpointResult.NotMatched[F]
-        else {
-          val req = input.request
-          EndpointResult.Matched(
-            input,
-            Trace.empty,
-            Effect[F].pure(Output.payload(decode(enumeratorFromReader(req.reader), req.charsetOrUtf8)))
-          )
+  implicit def enumeratorFromReader[F[_] : Effect]: StreamFromReader[Enumerator, F] =
+    StreamFromReader.instance { reader =>
+      def rec(reader: Reader[Buf]): Enumerator[F, Buf] = {
+        Enumerator.liftM[F, Option[Buf]] {
+          futureToEffect(reader.read())
+        }.flatMap {
+          case None => Enumerator.empty[F, Buf]
+          case Some(buf) => Enumerator.enumOne[F, Buf](buf).append(rec(reader))
         }
       }
-
-      final override def item: RequestItem = items.BodyItem
-      final override def toString: String = "enumeratorBody"
-  }
-
-  /**
-    * An evaluating [[Endpoint]] that reads a required chunked streaming JSON body, interpreted as
-    * an `Enumerator[Future, A]`. The returned [[Endpoint]] only matches chunked (streamed) requests.
-    */
-  def enumeratorJsonBody[F[_] : Effect, A](implicit

```

Comment:
Should we keep this around and deprecate so we can remove it next release?

Reply:
Yes, we can do so.

Reply:
Sorry, I don't see where we redefined them. Mind pointing me out? I believe we should keep these for the sake of backwards compatibility.

Reply:
I put it back